subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/logging/AbstractTestLogger.java

```diff
@@ -72,7 +72,7 @@ private String getEventPath(TestDescriptor descriptor) {
                 // level. This matters when configuring granularity.
                 names.add(current.getClassName() + "." + current.getName());
             } else {
-                names.add(current.getName());
+                names.add(current.getDisplayName());
```

Comment:
When I was looking at this file, I was wondering: does this change work if there is no display name?

Reply:
The `getDisplayName()` was not annotated with `@Nullable` so I assumed it is always present.